apps/prairielearn/src/pages/error/error.ts

```diff
@@ -35,6 +35,15 @@ export default (function (err, req, res, _next) {
     response_id: res.locals.response_id,
   });
 
+  // Check if the client only accepts JSON
+  if (req.accepts('application/json') && !req.accepts('html')) {
```

Comment:
This is actually really nice because then unexpected errors from the endpoint are bubbled correctly.

Reply:
Tangent, Rails has a cute little `respond_to` thing that I think is neat: https://github.com/rails/rails/blob/88ddc0e1c3ce62e9a9e217ae899099688fd6a176/actionpack/lib/action_controller/metal/mime_responds.rb#L211

Nothing immediately actionable here, just a nice API that avoids questions like mine here.